File: sdk-examples/python/streaming.py
"""Streaming Example — Watch tokens arrive.

Usage:
    python streaming.py
"""

from edgecrab import Agent


def main():
    agent = Agent("anthropic/claude-sonnet-4", streaming=True, quiet_mode=True)

    events = agent.stream_sync("Write a haiku about Python programming.")
    for event in events:
        if event.kind == "token":
            print(event.data, end="", flush=True)
        elif event.kind == "done":
            print("\n--- Done ---")
        else:
            print(f"[{event.kind}]: {event.data}")


if __name__ == "__main__":
    main()

File: sdk-examples/rust/streaming.rs
//! # Streaming Example — Watch tokens arrive in real-time
//!
//! ```bash
//! cargo run --example streaming
//! ```

use edgecrab_sdk::prelude::*;

#[tokio::main]
async fn main() -> Result<(), SdkError> {
    let agent = SdkAgent::builder("anthropic/claude-sonnet-4")?
        .streaming(true)
        .build()?;

    let mut rx = agent.stream("Write a haiku about Rust programming.").await?;

    while let Some(event) = rx.recv().await {
        match &event {
            StreamEvent::Token(token) => print!("{token}"),
            StreamEvent::Done => println!("\n--- Done ---"),
            other => println!("[event: {other:?}]"),
        }
    }

    Ok(())
}
